src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,259 @@
+ * from mutor => mute set, where "mute set" referrs to the set of cowns muted by
+ * a mutor. Each scheduler thread scans the mutors of its mute map on each
+ * iteration of the run loop. If any mutor is not muted and is not overloaded,
+ * the corresponding mute set is unmuted and the entry is removed from the mute
+ * map. A mutor may exist as the key in mutiple mute maps, but a muted cown may
+ * only be tracked in a single mute set.
+ *
+ * ## Message Scan
+ *
+ * All messages from one set of sending cowns to a set of receiving cowns is
+ * scanned to determine if the senders should be muted. Such a message will
+ * result in muted senders if all of the following are true:
+ *   1. The receiver set does not contain any of the senders
+ *   2. No sender is overloaded
+ *   3. One of the receivers is either overloaded or muted.
+ *
+ * A message is given "priority" if either condition 1 or 2 are false. A
+ * priority message will result in any muted receivers being unmuted so that the
+ * senders may make progress sooner.
+ *
+ * The backpressure bits of any cown may only be modified by the scheduler
+ * thread running the cown or holding the muted cown in its mute map. If a
+ * priority message requires a recipient cown to be unmuted, then the scheduler
+ * thread running the message will search for the cown in its mute map. If the
+ * cown does not exist, then it will be sent to other scheduler threads until it
+ * is found and subsequently unmuted.
+ *
+ * ## Cown Backpressure States
+ *
+ * These states are not explicitly encoded, but they are useful to consider.
+ *
+ *     +------------+     +------------+
+ *  +--|            |---->|            |--+
+ *  |  | Normal     |     | Muting     |  |
+ *  +->|            |<-+  |            |<-+
+ *     +------------+  ^  +------------+
+ *          | ^        |      |
+ *          v |        |      v
+ *     +------------+  |  +------------+
+ *     |            |  |  |            |
+ *     | Overloaded |  +--| Muted      |
+ *     |            |     |            |
+ *     +------------+     +------------+
+ *
+ */
+
+#include "ds/mpscq.h"
+
+namespace verona::rt
+{
+  class Backpressure
+  {
+    static constexpr uint32_t current_load_mask = 0x00'0000'ff;
+    static constexpr uint32_t load_hist_mask = 0x00'ffff'00;
+    static constexpr uint32_t muted_mask = 0x80'0000'00;
+    static constexpr uint32_t needs_token_mask = 0x40'0000'00;
+
+    static constexpr uint32_t overload_threshold = 800;
+    static constexpr uint32_t unmute_threshold = 100;
+
+    /**
+     * Backpressure bits contain the following fields:
+     *   [31:31] muted
+     *   [30:30] needs_token
+     *   [29:24] (unused)
+     *   [23:08] load_history
+     *   [07:00] current_load
+     *
+     * The `muted` bit indicates if a cown is currently muted. A muted cown must
+     * not run nor be collected until it has been unmuted.
+     *
+     * The `needs_token` bit indicates if the token message is not in a cown's
```

Comment:
how does this work? this is stuff that i don't know about and this makes me curious.

I think you need a firm understanding of the scheduler to understand this note. That might a reasonable thing to assume the reader has.